src/sysusers/sysusers.c

```diff
@@ -567,6 +571,17 @@ static int write_temporary_shadow(const char *shadow_path, FILE **tmpfile, char
                         return errno ? -errno : -EIO;
         }
 
+        /* Append the remaining NIS entries if any */
+        while (sp) {
+                errno = 0;
+                if (putspent(sp, shadow) < 0)
+                        return errno ? -errno : -EIO;
+
+                sp = fgetspent(original);
+        }
+        if (!IN_SET(errno, 0, ENOENT))
```

Comment:
If `sp` was NULL above, in line 575, then errno is not set by the loop and this check uses the value of errno inherited from somewhere above. I *think* it'll actually be set always, but this is very brittle. Please rework this to not use errno which is not set in the immediately preceding lines.

Reply:
Oh, I see that @poettering wrote the same thing while I was typing my comment ;)